hrui/js/src/civihr-popup/civihr-popup.js

```diff
@@ -4,14 +4,61 @@ Prevent Popups to overflow tables.
 (function ($) {
   'use strict';
 
-  // .btn-slide is a continer of popup and a button triggering it
-  $('.btn-slide').attrchange(function (e) {
-    var $body = $('body');
+  $('body').on('click', '.btn-slide', function (e) {
     var $button = $(this);
-    var $closestEntity, $popup, buttonOffset;
+    var $body = $('body');
+    var $popup, buttonOffset, $closestEntity;
+
+    e.preventDefault();
+
+    (function init () {
+      openPopupPanel();
+      listenToMouseOutEvent();
+    })();
+
+    /**
+     * Closes the popup panel and removes the .civihr-popup-open class from the
+     * body.
+     */
+    function closePopupPanel () {
+      $body
+        .children('.civihr-popup')
+        .appendTo($button)
+        .removeClass('civihr-popup');
+
+      $body.removeClass('civihr-popup-open');
+    }
 
-    // check if popup is open
-    if ($button.hasClass('btn-slide-active')) {
+    /**
+     * Listens to mouse events happening outside the of the panel in order to
+     * close it. The way to detected mouse out is by listening to the
+     * .btn-slide-active class change, which is added and removed by CiviCRM.
+     * the data *attrchange-is-on* is set to true in order to avoid adding
+     * multiple listeners to the same element.
+     */
+    function listenToMouseOutEvent () {
+      // AttrChange event is already listened, skip:
+      if ($button.data('attrchange-is-on')) {
+        return;
+      }
+
+      $button.data('attrchange-is-on', true);
+
+      $button.attrchange(function () {
+        // Button is already open, skip:
+        if ($button.hasClass('btn-slide-active')) {
+          return;
+        }
+
+        closePopupPanel();
+      });
+    }
+
+    /**
```

Comment:
I think it would be easier to read if you break the sentence like that:
```js
/**
 * Opens the popup panel
 * and adds the .civihr-popup-open class to the body
 */
```

Please note the change from "pop" to "popup" as well (for consistency)